optimade/models/entries.py

```diff
@@ -15,11 +15,33 @@
 )
 
 
+class TypedRelationship(Relationship):
```

Comment:
I would say that what we need it [this](https://docs.python.org/3.8/library/typing.html#typing.Literal). Unfortunately it is not introduced until Python 3.8